rebase: update csi-resizer to v1.13.1

nixpanic opened the pull request:
The Kubernetes CSI community has released an updated external-resizer
sidecar that fixes some issues and adds a few improvements.

See: https://github.com/kubernetes-csi/external-resizer/releases/tag/v1.13.1

nixpanic committed:
rebase: update csi-resizer to v1.13.1

The Kubernetes CSI community has released an updated external-resizer
sidecar that fixes some issues and adds a few improvements.

See: https://github.com/kubernetes-csi/external-resizer/releases/tag/v1.13.1
Signed-off-by: Niels de Vos <ndevos@ibm.com>

nixpanic commented:
/test ci/centos/k8s-e2e-external-storage/1.32

nixpanic commented:
/test ci/centos/mini-e2e-helm/k8s-1.30

nixpanic commented:
/test ci/centos/mini-e2e/k8s-1.31

nixpanic commented:
It seems the new container-image is not available in the registry yet.

nixpanic commented:
> It seems the new container-image is not available in the registry yet.

v1.13.0 failed to build and could not be pushed to the release repository, v1.13.1 contains the release-tools fixes and is now available.

nixpanic commented:
@mergifyio queue

nixpanic commented:
Interesting failure in the CI:
```
  [FAIL] External Storage [Driver: rbd.csi.ceph.com] [Testpattern: Dynamic PV (default fs)(allowExpansion)] volume-expand [It] Verify if offline PVC expansion works
```
Offline PVC expansion is not expected to work if the PVC has a filesystem on it :thinking:

Madhu-1 commented:
@nixpanic https://kubernetes.slack.com/archives/C09QZFCE5/p1736798710626099 it looks like its broken. why its running offline expansion for ceph-csi, should we disable it as we don't support it?

nixpanic commented:
> @nixpanic https://kubernetes.slack.com/archives/C09QZFCE5/p1736798710626099 it looks like its broken. why its running offline expansion for ceph-csi, should we disable it as we don't support it?

Yeah, I also wonder why it would run. Will look into it soonish. Maybe it is because `controllerExpansion: true` is set in `scripts/k8s-storage/driver-rbd.yaml`.

nixpanic committed:
ci: disable `offlineExpansion` in k8s-e2e-external-storage for RBD

The `offlineExpansion` option enables tests that try to expand volumes
in the CSI Controller (provisioner). This is only possible for BlockMode
volumes, as a FileMode volume needs run a filesystem resizer while the
volume is attached.

Signed-off-by: Niels de Vos <ndevos@ibm.com>

Madhu-1 commented on scripts/k8s-storage/driver-rbd.yaml:
```diff
@@ -50,8 +50,8 @@ DriverInfo:
-    # Support for volume expansion in controllers
-    controllerExpansion: true
+    # Support for volume expansion in controllers (only for RBD with BlockMode)
+    controllerExpansion: false
```
@nixpanic we still need this to set to true for filesystem as well as it needs both controllerExpansion and filesystem resize as well?

Madhu-1 replied:
https://github.com/kubernetes/kubernetes/blob/ec2e0de35a298363872897e5904501b029817af3/test/e2e/storage/framework/testdriver.go#L193 this need to be set to false as its enabled by default?

nixpanic replied:
yeah, the options are a bit confusing, and their documentation could be better... I now found an [offlineExpansion](https://github.com/kubernetes/kubernetes/blob/master/test/e2e/storage/framework/testdriver.go#L193) option, maybe that is more suitable.

nixpanic commented:
/test ci/centos/k8s-e2e-external-storage/1.32

nixpanic commented:
/test ci/centos/k8s-e2e-external-storage/1.32

Madhu-1 commented on scripts/k8s-storage/driver-rbd.yaml:
```diff
@@ -56,6 +56,9 @@ DriverInfo:
     nodeExpansion: true
 
+    # supports offline volume expansion
+    offlineExpansion: false
```
@nixpanic same need to be done for cephfs and nfs driver as well?

nixpanic replied:
no, CephFS and NFS can be expanded while the volume is not mounted.

Madhu-1 reviewed (approved):
LGTM

Rakshith-R commented:
@mergifyio queue
